catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -137,7 +143,27 @@ define([
             template: 'sourceList',
             itemView: SourceView.SourceRow,
             emptyView: EmptyView.sources,
-            itemViewContainer: 'tbody'
+            itemViewContainer: 'tbody',
+
+            initialize: function() {
+                this.hidePageLoading();
+                this.showSourcesLoading();
+            },
+            collectionEvents: {
+                "request": "showSourcesLoading",
+                "sync": "hideSourcesLoading"
+            },
+            showSourcesLoading: function() {
+                this.$el.addClass('hide');
+                $("#sources-loading").removeClass('hide');
```

Comment:
I think you should investigate making this model driven and avoid the use of global JQuery selectors that modify the DOM outside of the view's control.  You can use model events to trigger when to re-render on changes to the loading value in the model.

http://marionettejs.com/docs/v2.4.7/marionette.view.html#viewmodelevents-and-viewcollectionevents